databases.go

```diff
@@ -0,0 +1,249 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	databaseBasePath        = "/v2/databases"
+	databaseResizePath      = databaseBasePath + "/%s/resize"
+	databaseMigratePath     = databaseBasePath + "/%s/migrate"
+	databaseMaintenancePath = databaseBasePath + "/%s/maintenance"
+	databaseBackupsPath     = databaseBasePath + "/%s/backups"
+)
+
+// DatabasesService is an interface for interfacing with the databases endpoints
+// of the DigitalOcean API.
+// See: https://developers.digitalocean.com/documentation/v2#databases
+type DatabasesService interface {
+	List(context.Context, *ListOptions) ([]*Database, *Response, error)
+	Get(context.Context, string) (*Database, *Response, error)
+	Create(context.Context, *DatabaseCreateRequest) (*Database, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+	Resize(context.Context, string, *DatabaseResizeRequest) (*Response, error)
+	Migrate(context.Context, string, *DatabaseMigrateRequest) (*Response, error)
+	UpdateMaintenance(context.Context, string, *DatabaseUpdateMaintenanceRequest) (*Response, error)
+	ListBackups(context.Context, string, *ListOptions) ([]*DatabaseBackup, *Response, error)
```

Comment:
Do we want to support all endpoints including Pools, Replicas, Restore, Users and DBs? We can tag team these :)